### PR TITLE
Log more information when an exception occurs in table_service

### PR DESCRIPTION
**Pull request by AbelHu**

### CODE REVIEW

File: lib/azure/storage/table/table_service.rb [hunk 1]
```diff
@@ -106,6 +106,9 @@ def get_table(table_name, options={})
         response = call(:get, table_uri(table_name, query))
         results = Table::Serialization.hash_from_entry_xml(response.body)
         results[:updated]
+      rescue => e
+        raise e if response.nil?
+        raise "Response header: #{response.headers.inspect}\nResponse body: #{response.body.inspect}\n#{e.inspect}\n#{e.backtrace.join("\n")}"
```
Comment (yaxia) [inline lib/azure/storage/table/table_service.rb]: How about putting these duplicated codes into one function?

**Comment by yaxia** [comment]
@AbelHu  Thanks for your contribution.
Could you submit the PR against the dev branch?
Could you add test cases (integration test) to cover your changes, especially when `raise e if response.nil?`?

**Comment by AbelHu** [comment]
@yaxia I will create a common function for those duplicated codes and create a new PR against the dev branch.
Do you have any suggestion for me to add unit test when response is nil?

**Comment by yaxia** [comment]
Regarding the test for nil response, it could happened when the connection cannot be established, e.g. wrong service endpoint set in the connection string or wrong resource URI.

**Comment by AbelHu** [comment]
@yaxia Sorry for the wrong condition. I think response = nil has been covered by current test cases as below. I need to add test case to cover that an exception is thrown when parsing the response. Do you have any idea about how to throw an exception when parsing the response?

```

    it "errors on an invalid table" do
      assert_raises(Azure::Core::Http::HTTPError) do
        subject.get_table "this_table.cannot-exist!"
      end
    end
```

**Comment by yaxia** [comment]
@AbelHu The PR turned to the master branch with your latest change. Could you set it to the dev branch?

**Comment by AbelHu** [comment]
Close this and will create a new one against dev branch.
